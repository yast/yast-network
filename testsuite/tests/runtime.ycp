{

// testedfiles: runtime.ycp Testsuite.ycp

include "testsuite.ycp";

map READ = $[
    "init" : $[
	"scripts" : $[ "exists" : true ]
    ],
    "probe" : $[
	"system" : [],
    ],
    "product" : $[
        "features" : $[
            "USE_DESKTOP_SCHEDULER" : "0",
            "ENABLE_AUTOLOGIN" : "0",
            "EVMS_CONFIG" : "0",
            "IO_SCHEDULER" : "cfg",
            "UI_MODE" : "expert",
        ],
    ],
    "sysconfig": $[
        "language" : $[
            "RC_LANG" : "",
            "DEFAULT_LANGUAGE" : "",
	    "ROOT_USES_LANG" : "no",
        ],
	"console" : $[
	    "CONSOLE_ENCODING" : "UTF-8",
	],
    ],
    "target" : $[
	"size" : 1,
        "stat" : $[ "dummy": true ]
    ],
];

map EXEC = $["target": $["bash_output": $[] ] ];

TESTSUITE_INIT([READ, $[], EXEC], nil);
include "network/runtime.ycp";

map EXEC0 = $[
    "target" : $[
	"bash" : 0,
	"bash_background" : 0,
	"bash_output" : $[
	    "exit"	: 0,
	    "stdout"	: "",
	    "stderr"	: "",
	],
    ],
];

map EXEC1 = $[
    "target" : $[
	"bash" : 1,
	"bash_background" : 1,
	"bash_output" : $[
	    "exit"	: 1,
	    "stdout"	: "",
	    "stderr"	: "Dummy error message",
	],
    ],
];

DUMP("RunSuSEconfig");
TEST(``(RunSuSEconfig()), [$[],$[],EXEC0], nil);
TEST(``(RunSuSEconfig()), [$[],$[],EXEC1], nil);

import "NetworkInterfaces";
NetworkInterfaces::Devices = $[ "dsl" : $[ "0" : $[] ]];

DUMP("SetupSMPPPD");
TEST(``(SetupSMPPPD(true)), [READ,$[],EXEC0], nil);
TEST(``(SetupSMPPPD(false)), [READ,$[],EXEC1], nil);

}
