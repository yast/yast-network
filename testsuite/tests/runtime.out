Dump	RunSuSEconfig
Execute	.target.bash "/sbin/SuSEconfig --module apache 2>/dev/null" 0
Execute	.target.bash "/sbin/SuSEconfig --module sendmail 2>/dev/null" 0
Return	true
Execute	.target.bash "/sbin/SuSEconfig --module apache 2>/dev/null" 1
Execute	.target.bash "/sbin/SuSEconfig --module sendmail 2>/dev/null" 1
Return	true
Dump	SetupSMPPPD
Return	true
Return	false
